pkg/oam/util/helper.go

```diff
@@ -197,3 +208,39 @@ func Object2Map(obj interface{}) (map[string]interface{}, error) {
 	err = json.Unmarshal(bts, &res)
 	return res, err
 }
+
+// GenTraitName generate trait name
+func GenTraitName(componentName string, ct *v1alpha2.ComponentTrait, collisionCount *int32) string {
+	return fmt.Sprintf("%s-%s-%s", componentName, TraitPrefixKey, ComputeHash(ct, collisionCount))
+}
+
+// ComputeHash returns a hash value calculated from pod template and
+// a collisionCount to avoid hash collision. The hash will be safe encoded to
+// avoid bad words.
+func ComputeHash(trait *v1alpha2.ComponentTrait, collisionCount *int32) string {
+	componentTraitHasher := fnv.New32a()
+	DeepHashObject(componentTraitHasher, *trait)
+
+	// Add collisionCount in the hash if it exists.
+	if collisionCount != nil {
+		collisionCountBytes := make([]byte, 8)
+		binary.LittleEndian.PutUint32(collisionCountBytes, uint32(*collisionCount))
```

Comment:
collisionCount is always passed in as new(int32) which I assume is a pointer to zero, I wonder why do we need this parameter?

Reply:
yes, I will remove this parameter.